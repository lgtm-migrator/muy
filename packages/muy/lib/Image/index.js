import React from "react"
import clsx from "clsx"
import makeStyles from "@material-ui/core/styles/makeStyles"
import imageStyle from "@muy/styles/lib/imageStyle";

const useStyles = makeStyles(imageStyle)

const Image = ({ alt, height, title, variant, width, ...props }) => {
    const classes = useStyles()
    return (
        <img
            alt={alt}
            className={clsx(classes.root, variant === "cover" && classes.cover)}
            height={height}
            title={title}
            width={width}
            {...props}
        />
    )
}

export default Image
