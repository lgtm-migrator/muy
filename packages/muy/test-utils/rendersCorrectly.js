import renderer from "react-test-renderer"

const rendersCorrectly = component => {
  const tree = renderer.create(component).toJSON()
  return expect(tree).toMatchSnapshot()
}

export default rendersCorrectly
