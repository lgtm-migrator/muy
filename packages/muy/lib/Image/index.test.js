import React from "react"
import rendersCorrectly from "../../test-utils/rendersCorrectly"
import Image from "."

describe("<Image/> component", () => {
  it("should render correctly", () => {
    rendersCorrectly(<Image />)
  })
})
