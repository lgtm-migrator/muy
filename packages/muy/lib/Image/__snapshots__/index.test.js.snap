// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Image component should render correctly with cover variant 1`] = `
<img
  className="makeStyles-root-1 makeStyles-cover-2"
/>
`;

exports[`Image component should render correctly with no props 1`] = `
<img
  className="makeStyles-root-1"
/>
`;
